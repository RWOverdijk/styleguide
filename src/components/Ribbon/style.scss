/** @define Ribbon */

@import "../../css/variables";

.Van-Ribbon {
  overflow: hidden;
  min-width: 250px;
  height: 47px;
  font-size: 18px;
  font-weight: $font-weight-bold;
  color: $color-green-dark;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  transform: rotate(45deg);
  background-color: $color-green-brightest;
}
