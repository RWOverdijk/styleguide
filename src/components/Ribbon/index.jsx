import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './style.scss';

class Ribbon extends PureComponent {
  render() {
    const { className, children, ...props } = this.props;

    return (
      <span className={classnames('Van-Ribbon', className)} {...props}>
        {children}
      </span>
    );
  }
}

Ribbon.defaultProps = {
  className: '',
  children: ''
};

Ribbon.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
};

export default Ribbon;
