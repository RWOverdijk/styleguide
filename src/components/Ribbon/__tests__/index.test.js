import React from 'react';
import renderer from 'react-test-renderer';
import Ribbon from '../';

describe("<Ribbon>", () => {
  it("renders default Ribbon", () => {
    const tree = renderer.create(
      <Ribbon>My Ribbon</Ribbon>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Ribbon className="my-class">My Ribbon</Ribbon>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Ribbon data-id="0">My Ribbon</Ribbon>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
