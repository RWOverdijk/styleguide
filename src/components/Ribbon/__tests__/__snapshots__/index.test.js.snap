// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon> renders default Ribbon 1`] = `
<span
  className="Van-Ribbon"
>
  My Ribbon
</span>
`;

exports[`<Ribbon> renders with specific attributes 1`] = `
<span
  className="Van-Ribbon"
  data-id="0"
>
  My Ribbon
</span>
`;

exports[`<Ribbon> renders with specific className 1`] = `
<span
  className="Van-Ribbon my-class"
>
  My Ribbon
</span>
`;
